toolchain/check/testdata/as/adapter_conversion.carbon

```diff
@@ -0,0 +1,447 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+// --- adapt_class.carbon
+
+library "adapt_class" api;
+
+class A {
+  var x: i32;
+  var y: i32;
+
+  fn Make() -> A {
+    return {.x = 1, .y = 2};
+  }
+}
+
+class B {
+  adapt A;
+}
+
+var a_ref: A = {.x = 1, .y = 2};
+let a_val: A = a_ref;
+
+let b_val: B = a_val as B;
+let b_ptr: B* = &(a_ref as B);
```

Comment:
Comment that `as` to an adapter type preserves category?

Reply:
Done.